File: Vanilla/sources/zdoc/media/lua/shared/Library/HairStyle.lua
---@class HairStyle : zombie.core.skinnedmodel.population.HairStyle
---@field public name String
---@field public model String
---@field public texture String
---@field public alternate ArrayList|Unknown
---@field public level int
---@field public trimChoices ArrayList|Unknown
---@field public growReference boolean
---@field public attachedHair boolean
---@field public noChoose boolean
HairStyle = {}

---@public
---@return boolean
function HairStyle:isValid() end

---@public
---@return boolean
function HairStyle:isAttachedHair() end

---@public
---@return String
function HairStyle:getName() end

---@public
---@return ArrayList|Unknown
function HairStyle:getTrimChoices() end

---@public
---@return int
function HairStyle:getLevel() end

---@public
---@param arg0 String
---@return String
function HairStyle:getAlternate(arg0) end

---@public
---@return boolean
function HairStyle:isNoChoose() end

---@public
---@return boolean
function HairStyle:isGrowReference() end

File: Vanilla/sources/zomboid/zombie/core/skinnedmodel/population/HairStyle.java
package zombie.core.skinnedmodel.population;

import java.util.ArrayList;
import javax.xml.bind.annotation.XmlAttribute;
import zombie.util.StringUtils;

public final class HairStyle {
   public String name = "";
   public String model;
   public String texture = "F_Hair_White";
   public final ArrayList alternate = new ArrayList();
   public int level = 0;
   public final ArrayList trimChoices = new ArrayList();
   public boolean growReference = false;
   public boolean attachedHair = false;
   public boolean noChoose = false;

   public boolean isValid() {
      return !StringUtils.isNullOrWhitespace(this.model) && !StringUtils.isNullOrWhitespace(this.texture);
   }

   public String getAlternate(String var1) {
      for(int var2 = 0; var2 < this.alternate.size(); ++var2) {
         HairStyle.Alternate var3 = (HairStyle.Alternate)this.alternate.get(var2);
         if (var1.equalsIgnoreCase(var3.category)) {
            return var3.style;
         }
      }

      return this.name;
   }

   public int getLevel() {
      return this.level;
   }

   public String getName() {
      return this.name;
   }

   public ArrayList getTrimChoices() {
      return this.trimChoices;
   }

   public boolean isAttachedHair() {
      return this.attachedHair;
   }

   public boolean isGrowReference() {
      return this.growReference;
   }

   public boolean isNoChoose() {
      return this.noChoose;
   }

   public static final class Alternate {
      @XmlAttribute
      public String category;
      @XmlAttribute
      public String style;
   }
}
